packages/itwinui-react/vitest.config.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: 'setupTests.ts',
+    coverage: {
+      provider: 'v8',
+    },
+    exclude: ['/node_modules/', '/esm/', '/cjs/'],
+    reporters: ['junit', 'default'],
+    outputFile: 'coverage/junit.xml',
+    sequence: {
+      hooks: 'list',
+    },
+  },
```

Comment:
vitest has a [`typecheck`](https://vitest.dev/config/#typecheck) option that we should enable and see how many tests fail. If it's only a few let's fix them right away; if there's more than a few, let's fix them in a separate PR

Also we should remove `test:types` in a future PR

Reply:
Yes. Basically all `test-types` files should be eventually removed. Type tests can be written more robustly using vitest. https://vitest.dev/guide/testing-types.html

Reply:
> Added ([5e40a96](https://github.com/iTwin/iTwinUI/commit/5e40a96fc6d92b25980c439d3bc43863ffa41a5c)). So no type tests fails as of right now.

I'm not sure. When I manually added a type error in `Table.test.tsx`, the tests still passed.

Reply:
Looks like it only checks `.test-d.tsx` files by default. We need to configure it using [`typecheck.include`](https://vitest.dev/config/#typecheck-include)

Reply:
Only `test.tsx`, because `test-types.tsx` is completely unrelated and being handled by `tsc` explicitly.

```js
include: ['**/*.test.?(c|m)[jt]s?(x)']
```